src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationapi/integration/OAuthIntTest.kt

```diff
@@ -377,19 +377,84 @@ class OAuthIntTest : IntegrationTestBase() {
 
       val token = getTokenPayload(String(tokenResponse))
       assertThat(token.get("authorities").toString()).isEqualTo(JSONArray(listOf("ROLE_TESTING", "ROLE_MORE_TESTING")).toString())
-      assertThat(token.get("sub")).isEqualTo("username")
+      assertThat(token.get("sub")).isEqualTo("name")
       assertThat(token.get("aud")).isEqualTo(validClientId)
 
       assertThat(token.get("client_id")).isEqualTo(validClientId)
       assertThat(token.get("grant_type")).isEqualTo(GrantType.authorization_code.name)
-      assertThat(token.get("auth_source")).isEqualTo(AuthSource.Auth.source)
+      assertThat(token.get("auth_source")).isEqualTo(AuthSource.Auth.name)
       assertThat(token.get("scope").toString()).isEqualTo(JSONArray(listOf("read")).toString())
       assertThat(token.get("user_id")).isEqualTo("9999")
       assertThat(token.get("name")).isEqualTo("name")
       assertThat(fullJsonResponse.get("user_name")).isEqualTo("username")
       assertThat(token.get("user_uuid")).isEqualTo("1234-5678-9999-1111")
     }
 
+    @Test
+    fun `test jwt fields in authorization-code flow`() {
+      validClientId = "test-auth-code-client-with-jwt-settings"
```

Comment:
Would be safer to hard code this client id in the uri rather than re-set a class field